src/TestUtils/src/Microsoft.Maui.IntegrationTests/TemplateTests.cs

```diff
@@ -113,6 +113,55 @@ public void BuildUnpackaged(string id, string framework, string config)
 				$"Project {Path.GetFileName(projectFile)} failed to build. Check test output/attachments for errors.");
 		}
 
+		[Test]
+		[TestCase("maui", $"{DotNetCurrent}-ios", "ios-arm64")]
+		public void PublishNativeAOT(string id, string framework, string runtimeIdentifier)
+		{
+			if (!TestEnvironment.IsMacOS)
+				Assert.Ignore("Publishing a MAUI iOS app with NativeAOT is only supported on a host MacOS system.");
+
+			var projectDir = TestDirectory;
+			var projectFile = Path.Combine(projectDir, $"{Path.GetFileName(projectDir)}.csproj");
+
+			Assert.IsTrue(DotnetInternal.New(id, projectDir, DotNetCurrent),
+				$"Unable to create template {id}. Check test output for errors.");
+
+			var extendedBuildProps = BuildProps;
+			extendedBuildProps.Add("PublishAot=true");
+			extendedBuildProps.Add("PublishAotUsingRuntimePack=true");  // TODO: This parameter will become obsolete https://github.com/dotnet/runtime/issues/87060
+
+			Assert.IsTrue(DotnetInternal.Publish(projectFile, "Release", framework: framework, properties: extendedBuildProps, runtimeIdentifier: runtimeIdentifier),
+				$"Project {Path.GetFileName(projectFile)} failed to build. Check test output/attachments for errors.");
+		}
+
+		[Test]
+		[TestCase("maui", $"{DotNetCurrent}-ios", "ios-arm64")]
+		public void PublishNativeAOTCheckWarnings(string id, string framework, string runtimeIdentifier)
```

Comment:
@ivanpovazan seems this is failing on this branch, for net9.0 can you take a look ?

Reply:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8817979&view=ms.vss-test-web.build-test-results-tab&runId=90608599&resultId=100050&paneView=debug

Expected warnings file path 'ILC' was not found.
Expected: not null
But was:  null

Reply:
Please see my comment at: https://github.com/dotnet/maui/pull/19441/files/f7ac0366ac299b1486b26ebada4a51b66a3d846d#r1428262698